packages/makeswift/src/integrate-next-app.ts

```diff
@@ -214,16 +216,26 @@ export default function Page({ snapshot }: Props) {
   const catchAllRoute = generateCatchAllRoute(isTypeScript)
   const extension = isTypeScript ? 'ts' : 'js'
 
-  // @todo: need to detect if a catch all route already exists
-  const catchAllRouteFilename =
-    fs.existsSync(path.join(pagesFolder, `index.${extension}`)) ||
-    fs.existsSync(path.join(pagesFolder, `index.${extension}x`))
-      ? `[...path].${extension}x`
-      : `[[...path]].${extension}x`
-
   // catch-all-route does not exist
-  if (glob.sync(path.join(pagesFolder, `\\[*\\].${extension}*`)).length === 0) {
-    fs.writeFileSync(path.join(temporaryPagesFolder, catchAllRouteFilename), catchAllRoute)
+  if (glob.sync(path.join(pagesFolder.absolute, `\\[*\\].${extension}*`)).length === 0) {
```

Comment:
What's this glob checking? If any sort of dynamic route exists? This check might be a bit too restrictive since there might be a dynamic route but it might not be a catch-all route. To check if it's a catch-all, we'd want to check if any file starts with `[[...` or `[...`

Reply:
We also don't have to worry if it's nested inside a folder since those catch-all routes won't conflict so a regular expression or a simple `startsWith` might be enough here

Reply:
Yeah I agree with that, I think when I wrote this I thought `/foo/bar/baz/[[...slug]].tsx` — would conflict with us somehow.

Thinking now, it won't.

Reply:
- it only checks the pages folder, no sub-folders
- you're right about dynamic routes that are not catch-all routes — do we want to not catch that? Like if they have `pages/[slug].tsx`? Would Next then only match things like `/foo/bar` for our Makeswift page?

If we want to find catch-all routes that are not also top level dynamic routes, we could use `glob.sync("\\[?(\\[)...*\\]?(\\]).*")`

And with `startsWith` or a regex — I'd also be fine with a regex (I do think they're cleaner than globs), but we also need something that searches recursively through all files. We're not just matching a string

Reply:
If we keep the glob as is, I should change the error message to say "we detected a dynamic route"

Reply:
> you're right about dynamic routes that are not catch-all routes — do we want to not catch that? Like if they have pages/[slug].tsx? Would Next then only match things like /foo/bar for our Makeswift page?

Yeah, that's correct. This is actually the setup Recurrency has. They have `[slug].tsx` and `[[...path]].tsx`. The former does WordPress pages and the latter Makeswift pages. They're also slowly making `[slug].tsx` pull pages from Makeswift as they incrementally move over.

> And with startsWith or a regex — I'd be fine with a regex (I do think they're cleaner than globs), but we also need something that iterates through the files in the pages folder. We're not just matching a string. Glob does that nicely, in my opinion

Ah, yes, we'd have to list the top-level files in the `pages` directory. Up to you whether you wanna use a glob or a regular expression, etc.

> If we keep the glob as is, I should change the error message to say "we detected a dynamic route"

Sounds good.

I think that it's probably safer to catch _any_ dynamic route at the top level of `pages` even if something like `[slug].tsx` wouldn't conflict. We generate paths at the top-level in the builder so it would be confusing if we integrated but things just didn't work. I'd rather they do things manually or reach out to us at that point.

Reply:
Ok cool. I will check for any top level dynamic route (the current glob), and change the error message to be clear